tests/std/tests/P0448R4_iosfwd/test.cpp

```diff
@@ -0,0 +1,123 @@
+#endif // _HAS_CXX23
+
+    static void test_forward_declaration(filebuf*);
+    static void test_forward_declaration(ifstream*);
+    static void test_forward_declaration(ofstream*);
+    static void test_forward_declaration(fstream*);
+
+    static void test_forward_declaration(wfilebuf*);
+    static void test_forward_declaration(wifstream*);
+    static void test_forward_declaration(wofstream*);
+    static void test_forward_declaration(wfstream*);
+
+#if _HAS_CXX20
+    static void test_forward_declaration(syncbuf*);
+    static void test_forward_declaration(osyncstream*);
+
+    static void test_forward_declaration(wsyncbuf*);
+    static void test_forward_declaration(wosyncstream*);
+#endif // _HAS_CXX20
+
+#if defined(_CRTBLD)
+    static void test_forward_declaration(ushistream*);
+    static void test_forward_declaration(ushostream*);
+    static void test_forward_declaration(ushfilebuf*);
+#endif // defined(_CRTBLD)
+};
+
+template <class CharT>
+struct test_forward_declarations {
+    static void test_forward_declaration(allocator<CharT>*);
+
+    static void test_forward_declaration(char_traits<CharT>*);
+    static void test_forward_declaration(istreambuf_iterator<CharT>*);
+    static void test_forward_declaration(ostreambuf_iterator<CharT>*);
+
+    static void test_forward_declaration(basic_ios<CharT>*);
+
+    static void test_forward_declaration(basic_streambuf<CharT>*);
+    static void test_forward_declaration(basic_istream<CharT>*);
+    static void test_forward_declaration(basic_ostream<CharT>*);
+    static void test_forward_declaration(basic_iostream<CharT>*);
+
+    static void test_forward_declaration(basic_stringbuf<CharT>*);
+    static void test_forward_declaration(basic_istringstream<CharT>*);
+    static void test_forward_declaration(basic_ostringstream<CharT>*);
+    static void test_forward_declaration(basic_stringstream<CharT>*);
+
+#if _HAS_CXX23
+    static void test_forward_declaration(basic_spanbuf<CharT>*);
+    static void test_forward_declaration(basic_ispanstream<CharT>*);
+    static void test_forward_declaration(basic_ospanstream<CharT>*);
+    static void test_forward_declaration(basic_spanstream<CharT>*);
+#endif // _HAS_CXX23
+
+    static void test_forward_declaration(basic_filebuf<CharT>*);
+    static void test_forward_declaration(basic_ifstream<CharT>*);
+    static void test_forward_declaration(basic_ofstream<CharT>*);
+    static void test_forward_declaration(basic_fstream<CharT>*);
+
+#if _HAS_CXX20
+    static void test_forward_declaration(basic_syncbuf<CharT>*);
+    static void test_forward_declaration(basic_osyncstream<CharT>*);
+#endif // _HAS_CXX20
+};
+
+int main() {
+    test_aliases aliases;
+    (void) aliases;
+
+    test_forward_declarations<char> forward_declarations_char;
+    (void) forward_declarations_char;
+
+    test_forward_declarations<wchar_t> forward_declarations_wchar_t;
+    (void) forward_declarations_wchar_t;
+}

```

Comment:
This isn't executing any code at runtime, so I believe that the test should be named `test.compile.pass.cpp`, we should have another function instantiating these classes, and then `main()` should be empty with the conventional compile-only comment:

https://github.com/microsoft/STL/blob/bd7adb4a932725f60ba096580c415616486ab64c/tests/std/tests/VSO_0180469_ptr_cat/test.compile.pass.cpp#L25